mcstatus/server.py

```diff
@@ -150,25 +150,33 @@ async def _retry_async_status(self, connection: TCPAsyncSocketConnection, **kwar
         result = await pinger.read_status()
         return result
 
-    def query(self) -> QueryResponse:
-        """Checks the status of a Minecraft Java Edition server via the query protocol."""
+    def query(self, tries: int = 3) -> QueryResponse:
+        """Checks the status of a Minecraft Java Edition server via the query protocol.
+
+        :param tries: The number of times to retry if an error is encountered.
+        :return: Query information in a :class:`~mcstatus.querier.QueryResponse` instance.
+        """
         ip = str(self.address.resolve_ip())
-        return self._retry_query(Address(ip, self.address.port))
+        return self._retry_query(Address(ip, self.address.port), tries=tries)
 
     @retry(tries=3)
-    def _retry_query(self, addr: Address) -> QueryResponse:
+    def _retry_query(self, addr: Address, **_kwargs) -> QueryResponse:
```

Comment:
I realize it looks silly to have an unused argument, but `Callable.kwargs` is unmodifiable for now. This PEP mentions that support may be added in the future https://peps.python.org/pep-0612/#concatenating-keyword-parameters
And until that's the case, there is not a way for typing tools like pyright to know that our retry decorator pops a keyword argument out of the `Callable`.

Reply:
I think it could be done with Concatenate if we make the `tries` argument required when using the decorator (see my approval comment for why we might actually want to do this). But I'm not sure pyright detects this, as the unused argument logic might be separated from that, so it's possible it would trigger unused warnings anyway.

Another way to solve it would be to make the decorator check for `_tries` instead of `tries`, and only make the user-facing (non-retry) functions take `tries`, but pass them over as `_tries=tires` when calling the retry func.

Reply:
That PEP talks about Concatenate making sense for modifying keyword arguments but follows up with why it will be tricky to implement. `Callable.kwargs` is not editable yet for the reasons outlined in the PEP.

Reply:
Oh right, kwargs aren't supported by `Concatenate`. I assumed the pep jut pointed to param specs as a whole instead of actually reading the linked section. Hm, that's annoying. In this case only the `_tries` suggestion makes sense here.